hi-ml-cpath/testhisto/testhisto/utils/test_plots_utils.py

```diff
@@ -72,19 +74,20 @@ def test_plots_handler_plots_only_desired_plot_options(plot_options: Collection[
     tiles_selector.top_slides_heaps = {0: [slide_node] * n_tiles, 1: [slide_node] * n_tiles}
     tiles_selector.bottom_slides_heaps = {0: [slide_node] * n_tiles, 1: [slide_node] * n_tiles}
 
-    with patch("health_cpath.utils.plots_utils.save_slide_thumbnail_and_heatmap") as mock_slide:
-        with patch("health_cpath.utils.plots_utils.save_top_and_bottom_tiles") as mock_tile:
-            with patch("health_cpath.utils.plots_utils.save_scores_histogram") as mock_histogram:
-                with patch("health_cpath.utils.plots_utils.save_confusion_matrix") as mock_conf:
-                    with patch("health_cpath.utils.plots_utils.save_pr_curve") as mock_pr:
-                        plots_handler.save_plots(
-                            outputs_dir=MagicMock(), tiles_selector=tiles_selector, results=MagicMock()
-                        )
+    with patch("health_cpath.utils.plots_utils.save_slide_thumbnail") as mock_slide:
```

Comment:
[Minor] These nested context managers are getting pretty unwieldy... One idea to make it cleaner could be to create a dictionary of patches, indexed by `PlotOption`s. Something like this?
```python
patchers = {
    PlotOption.SLIDE_THUMBNAIL: patch("health_cpath.utils.plots_utils.save_slide_thumbnail"),
    ...
}

mock_funcs = {option: patcher.start() for option, patcher in patchers}
...
patch.stopall()

calls_count = 0
for option, mock_func in mock_funcs.items():
    calls_count += assert_plot_func_called_if_among_plot_options(mock_func, option, plot_options)
```

Reply:
Thanks for the suggestion 😉